extensions/interactions/CodeRepl/directives/oppia-interactive-code-repl.component.ts

```diff
@@ -0,0 +1,292 @@
+      )
+    );
+    const {
+      language,
+      placeholder,
+      preCode,
+      postCode
+    } = this.interactionAttributesExtractorService.getValuesFromAttributes(
+      'CodeRepl',
+      {
+        languageWithValue: this.languageWithValue,
+        placeholderWithValue: this.placeholderWithValue,
+        preCodeWithValue: this.preCodeWithValue,
+        postCodeWithValue: this.postCodeWithValue,
+      }
+    ) as CodeReplCustomizationArgs;
+    this.interactionIsActive = (this.lastAnswer === null);
+    this.language = language.value;
+    this.placeholder = placeholder.value;
+    this.preCode = preCode.value;
+    this.postCode = postCode.value;
+
+    // Make sure this.preCode ends with a newline.
+    if (this.preCode.trim().length === 0) {
+      this.preCode = '';
+    } else if (this.preCode.slice(-1) !== '\n') {
+      this.preCode += '\n';
+    }
+
+    // Make sure this.placeholder ends with a newline.
+    if (this.placeholder.slice(-1) !== '\n') {
+      this.placeholder += '\n';
+    }
+
+    this.hasLoaded = false;
+
+    // Keep the code string given by the user and the stdout from the
+    // evaluation until sending them back to the server.
+    if (this.interactionIsActive) {
+      this.code = (
+        this.preCode + this.placeholder + this.postCode);
+      this.output = '';
+    } else {
+      this.code = this.lastAnswer().code;
+      this.output = this.lastAnswer().output;
+    }
+
+    // Configure Skulpt.
+    Sk.configure({
+      output: (out) => {
+        // This output function is called continuously throughout the
+        // runtime of the script.
+        this.output += out;
+      },
+      read: (name) => {
+        // This function is called when a builtin module is imported.
+        if (Sk.builtinFiles.files[name] === undefined) {
+          // If corresponding module is not present then,
+          // removal of this block also results in failure of import.
+          throw new Error('module ' + name + ' not found');
+        }
+        return Sk.builtinFiles.files[name];
+      },
+      timeoutMsg: () => {
+        this.sendResponse('', 'timeout');
+      },
+      execLimit: 10000
+    });
+
+    this.currentInteractionService.registerCurrentInteraction(
+      () => this.submitAnswer(), null);
+  }
+
+  ngAfterViewInit(): void {
+    const runAfterTimeout = () => {
+      this.initMarkers(this.codeMirrorComponent.codeMirror);
+      this.hasLoaded = true;
+      this.changeDetectionRef.detectChanges();
+    };
+    setTimeout(() => {
+      runAfterTimeout();
+    }, 0);
+  }
+
+  runAndSubmitCode(codeInput: string): void {
+    this.runCode(codeInput, (evaluation, err) => {
+      this.sendResponse(evaluation, err);
+    });
+  }
+  private submitAnswer() {
+    this.runAndSubmitCode(this.code);
+  }
+
+  onEditorChange(code: string): void {
+    this.code = code;
+  }
+
+  runCode(
+      codeInput: string, onFinishRunCallback: (
+        x: string, error: string) => void): void {
```

Comment:
This is a bit hard to read, can we do?
```suggestion
      codeInput: string, 
      onFinishRunCallback: (x: string, error: string) => void): void {
```